### ISSUE TITLE
Pod::Usage gives warnings when -exitval => "NOEXIT"

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6438]
Migrated from [rt.perl.org#21944](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=21944) (status was 'resolved')

Searchable as RT21944$

**Comment by p5pRT** [comment]
### From @sciurius

#### Created by @sciurius



Pod&#8203;::Usage allows option \-exitval to be the string "NOEXIT"\, but in
this case it gives one or more warnings since the value is used in
numeric comparisons\.

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=medium

Site configuration information for perl v5.8.0:

Configured by jv at Fri Jul 19 10:07:18 CEST 2002.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.9-34, archname=i386-linux-thread-multi
    uname='linux phoenix.squirrel.nl 2.4.9-34 #1 sat jun 1 06:10:23 edt 2002 i686 unknown '
    config_args='-des -Darchname=i386-linux -Dprefix=/opt/perl-5.8.0 -Dcf_email=jvromans@squirrel.nl -Dperladmin=jvromans@squirrel.nl -Dusedevel=true -Uversiononly -Dusethreads -Duseithreads'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/gdbm',
    optimize='-O2',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fno-strict-aliasing -I/usr/local/include -I/usr/include/gdbm'
    ccversion='', gccversion='2.96 20000731 (Red Hat Linux 7.1 2.96-98)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lnsl -lndbm -lgdbm -ldl -lm -lpthread -lc -lcrypt -lutil
    perllibs=-lnsl -ldl -lm -lpthread -lc -lcrypt -lutil
    libc=/lib/libc-2.2.4.so, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version='2.2.4'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fpic', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.0:
    /opt/perl-5.8.0/lib/5.8.0/i386-linux-thread-multi
    /opt/perl-5.8.0/lib/5.8.0
    /opt/perl-5.8.0/lib/site_perl/5.8.0/i386-linux-thread-multi
    /opt/perl-5.8.0/lib/site_perl/5.8.0
    /opt/perl-5.8.0/lib/site_perl
    .


Environment for perl v5.8.0:
    HOME=/home/jv
    LANG=en_US
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/opt/perl-5.8.0/bin:.:/home/jv/bin:/usr/local/bin:/bin:/usr/bin:/usr/X11R6/bin
    PERL_BADLANG (unset)
    SHELL=/bin/tcsh

```
</details>

**Comment by p5pRT** [comment]
### From mark.addison@itn.co.uk

Hello\,

\(Is this the right place or should I mail the author direct?\)

Bug


---
Perl 5\.8\.0
Pod&#8203;::Usage 1\.14 

The 'NOEXIT' value for the \-exitval option to pod2usage doesn't work as
documented\. Code uses 'noexit'\. Also emits warnings when used;

 Argument "noexit" isn't numeric in numeric lt \(\<\) at    
 /usr/local/lib/perl5/5\.8\.0/Pod/Usage\.pm line 472\.

Patch


---
Sorry\, I can't do context diffs on this box\.

$ diff \-c Usage\.pm\.orig Usage\.pm
\*\*\* Usage\.pm\.orig       Tue Aug  5 12&#8203;:16&#8203;:39 2003
\-\-\- Usage\.pm    Tue Aug  5 12&#8203;:26&#8203;:39 2003
\*\*\*\*\*\*\*\*\*\*\*\*\*\*\*
\*\*\* 467\,476 \*\*\*\*
&nbsp;      elsif \(\! defined $opts\{"\-verbose"\}\) \{
&nbsp;          $opts\{"\-verbose"\} = \($opts\{"\-exitval"\} \< 2\);
&nbsp;      \}

&nbsp;      \#\# Default the output file
\!     $opts\{"\-output"\} = \($opts\{"\-exitval"\} \< 2\) ? \\\*STDOUT : \\\*STDERR
\!             unless \(defined $opts\{"\-output"\}\);
&nbsp;      \#\# Default the input file
&nbsp;      $opts\{"\-input"\} = $0  unless \(defined $opts\{"\-input"\}\);

\-\-\- 467\,478 \-\-\-\-
&nbsp;      elsif \(\! defined $opts\{"\-verbose"\}\) \{
&nbsp;          $opts\{"\-verbose"\} = \($opts\{"\-exitval"\} \< 2\);
&nbsp;      \}
\+       $opts\{"\-exitval"\} = lc $opts\{"\-exitval"\};  \# Normalize 'noexit'
value

&nbsp;      \#\# Default the output file
\!     $opts\{"\-output"\} = \($opts\{"\-exitval"\} ne 'noexit' and
$opts\{"\-exitval"\} \< 2\)
\!                                        ? \\\*STDOUT : \\\*STDERR
\!                      unless \(defined $opts\{"\-output"\}\);
&nbsp;      \#\# Default the input file
&nbsp;      $opts\{"\-input"\} = $0  unless \(defined $opts\{"\-input"\}\);



mark
"In the beginning\, there was nothing\, which exploded\."





This email \(and any attachments\) is intended solely for the individual\(s\) to
whom addressed\. It may contain confidential and/or legally privileged
information\. Any statement or opinions therein are not necessarily those of
ITN unless specifically stated\. Any unauthorised use\, disclosure or copying
is prohibited\. If you have received this email in error\, please notify the
sender and delete it from your system\. Security and reliability of the
e\-mail and attachments are not guaranteed\. You must take full responsibility
for virus checking\.

Independent Television News Limited\, 
Registered No\. 548648 England\,
VAT Reg\. No&#8203;: GB 756 2995 81\, 
200 Gray's Inn Road\, London WC1X 8XZ\,
Telephone&#8203;: 0207 833 3000\.

**Comment by p5pRT** [comment]
### From @rspier

From&#8203;: "John M\. Gamble" \<jgamble@&#8203;ripco\.com> 
Subject&#8203;: Re&#8203;: Perl5 Bug Summary \(Jan 12th message on perl5\-porters
mailing list \\
archive\) 
Date&#8203;: Tue\, 20 Jan 2004 13&#8203;:52&#8203;:13 \-0600 
 
I just did a quick scan of the subject lines\, so I may be wrong\, but 
isn't number 23224 \("noexit Bug in Pod&#8203;::Usage\. \[Patch\]"\) actually a fix 
for number 21944 \("Pod&#8203;::Usage gives warnings when \-exitval => 
"NOEXIT""\)? 
 
&nbsp;        \-john

**Comment by p5pRT** [comment]
@rspier - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @smpeters

> \[robert \- Tue Jan 20 13&#8203;:09&#8203;:53 2004\]&#8203;:
> 
> From&#8203;: "John M\. Gamble" \<jgamble@&#8203;ripco\.com> 
> Subject&#8203;: Re&#8203;: Perl5 Bug Summary \(Jan 12th message on perl5\-porters
> mailing list \\
> archive\) 
> Date&#8203;: Tue\, 20 Jan 2004 13&#8203;:52&#8203;:13 \-0600 
>  
> I just did a quick scan of the subject lines\, so I may be wrong\, but 
> isn't number 23224 \("noexit Bug in Pod&#8203;::Usage\. \[Patch\]"\) actually a fix 
> for number 21944 \("Pod&#8203;::Usage gives warnings when \-exitval => 
> "NOEXIT""\)? 
>  

This fix was applied with change \#19990\.

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
